DataFormats/Detectors/GlobalTracking/include/DataFormatsGlobalTracking/RecoContainerCreateTracksVariadic.h

```diff
@@ -0,0 +1,164 @@
+  // 3) For tracks whose timing is provided as RO frame: as time in \mus for RO frame start since the start of TF, half-duration of RO window and 0.
+
+  auto start_time = std::chrono::high_resolution_clock::now();
+  constexpr float PS2MUS = 1e-6;
+  std::array<std::vector<uint8_t>, GTrackID::NSources> usedData;
+  auto flagUsed2 = [&usedData](int idx, int src) {
+    if (!usedData[src].empty()) {
+      usedData[src][idx] = 1;
+    }
+  };
+  auto flagUsed = [&usedData, &flagUsed2](const GTrackID gidx) { flagUsed2(gidx.getIndex(), gidx.getSource()); };
+  auto isUsed2 = [&usedData](int idx, int src) { return (!usedData[src].empty()) && (usedData[src][idx] != 0); };
+  auto isUsed = [&usedData, isUsed2](const GTrackID gidx) { return isUsed2(gidx.getIndex(), gidx.getSource()); };
+
+  // create only for those data types which are used
+  const auto& tracksITS = getITSTracks<o2::its::TrackITS>();
+  const auto& tracksTPC = getTPCTracks<o2::tpc::TrackTPC>();
+  const auto& tracksTPCITS = getTPCITSTracks<o2::dataformats::TrackTPCITS>();
+  const auto& tracksTPCTOF = getTPCTOFTracks<o2::dataformats::TrackTPCTOF>();
+  const auto& matchesTPCTOF = getTPCTOFMatches<o2::dataformats::MatchInfoTOF>();
+
+  usedData[GTrackID::ITS].resize(tracksITS.size());                                      // to flag used ITS tracks
+  usedData[GTrackID::TPC].resize(tracksTPC.size());                                      // to flag used TPC tracks
+  usedData[GTrackID::ITSTPC].resize(tracksTPCITS.size());                                // to flag used ITSTPC tracks
+  usedData[GTrackID::TOF].resize(getTOFMatches<o2::dataformats::MatchInfoTOF>().size()); // to flag used ITSTPC-TOF matches
+
+  // ITS-TPC-TOF matches, may refer to ITS-TPC (TODO: something else?) tracks
+  {
+    auto matches = getTOFMatches<o2::dataformats::MatchInfoTOF>(); // thes are just MatchInfoTOF objects, pointing on ITS-TPC match and TOF cl.
+    auto tofClusters = getTOFClusters<o2::tof::Cluster>();
+    if (matches.size() && (!tofClusters.size() || !tracksTPCITS.size())) {
+      throw std::runtime_error(fmt::format("Global-TOF tracks ({}) require ITS-TPC tracks ({}) and TOF clusters ({})",
+                                           matches.size(), tracksTPCITS.size(), tofClusters.size()));
+    }
+    for (unsigned i = 0; i < matches.size(); i++) {
+      const auto& match = matches[i];
+      const auto& tofCl = tofClusters[match.getTOFClIndex()];
+      float timeTOFMUS = (tofCl.getTime() - match.getLTIntegralOut().getTOF(o2::track::PID::Pion)) * PS2MUS; // tof time in \mus, FIXME: account for time of flight to R TOF
+      const float timeErr = 0.010f;                                                                          // assume 10 ns error FIXME
+
+      auto gidx = match.getEvIdxTrack().getIndex(); // this should be corresponding ITS-TPC track
+      if (creator(tracksPool.get(gidx), {i, GTrackID::ITSTPCTOF}, timeTOFMUS, timeErr)) {
```

Comment:
my old creator was anyway accepting TrackParCov only, in this version it should be 
````
if (creator(tracksPool.get_as<o2::dataformats::TrackTPCITS>(gidx), {i, GTrackID::ITSTPCTOF}, timeTOFMUS, timeErr)) {
````
since the ITSTPCTOF is just a pair of references on TrackTPCITS and TOF cluster.